kong/db/schema/init.lua

```diff
@@ -1599,6 +1610,33 @@ function Schema:validate(input, full_check)
 end
 
 
+-- Iterate through input fields on update and check agianst schema for 
+-- immutable attribute. If immutable attribute is set, compare input values
+-- against entity values to detirmine whether input is valid.
+-- @param input The input table.
+-- @param entity The entity update will be performed on.
+-- @return True on success.
+-- On failure, it returns nil and a table containing all errors by field name.
+-- In all cases, the input table is untouched.
+function Schema:validate_immutable_fields(input, entity)
+  local errors = {}
+
+  input = tablex.deepcopy(input)
```

Comment:
Any reason for deep-copying this table (a fairly expensive operation) since we are not modifying its values?